dsb-integration/src/test/java/com.spectralogic.dsbrowser.integration/tasks/Ds3DeleteFolderTaskTest.java

```
@@ -0,0 +1,122 @@
+                    "80",
+                    null,
+                    new SavedCredentials(
+                            client.getConnectionDetails().getCredentials().getClientId(),
+                            client.getConnectionDetails().getCredentials().getKey()),
+                    false,
+                    false);
+            session = CreateConnectionTask.createConnection(SessionModelService.setSessionModel(savedSession, false), resourceBundle, buildInfoService);
+            try {
+                envDataPolicyId = IntegrationHelpers.setupDataPolicy(TEST_ENV_NAME, false, ChecksumType.Type.MD5, client);
+                envStorageIds = IntegrationHelpers.setup(TEST_ENV_NAME, envDataPolicyId, client);
+                HELPERS.ensureBucketExists(bucketName, envDataPolicyId);
+                latch.countDown();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        });
+        latch.await();
+    }
+
+    @After
+    public void teardown() throws IOException {
+        IntegrationHelpers.teardown(TEST_ENV_NAME, envStorageIds, client);
+        client.close();
+    }
+
+    @Test
+    public void deleteFolder() throws Exception {
+        HELPERS.createFolder(bucketName, folderName);
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        Platform.runLater(() -> {
+            try {
+                final Ds3DeleteFolderTask deleteFolderTask = new Ds3DeleteFolderTask(session.getClient(), bucketName, folderName);
+                workers.execute(deleteFolderTask);
+                deleteFolderTask.setOnSucceeded(event -> {
+                    successFlag = true;
+                    latch.countDown();
+                });
+                deleteFolderTask.setOnFailed(event -> latch.countDown());
+                deleteFolderTask.setOnCancelled(event -> latch.countDown());
+            } catch (final Exception e) {
+                e.printStackTrace();
```

If this is a legitimate error condition, then let it fall out to the junit runner instead of handling it here, and swallowing the exception.